library/Zend/Mail/Transport/Smtp.php

```diff
@@ -76,6 +81,28 @@ public function getOptions()
     {
         return $this->options;
     }
+    
+    /**
+     * Set options
+     *
+     * @param  SmtpOptions $options
+     * @return Smtp
+     */
+    public function setEnvelope(Envelope $envelope)
+    {
+        $this->envelope = $envelope;
+        return $this;
```

Comment:
That's quite ok for now. Just the return value of `setEnvelope` should be `void`